### DISCUSSION TITLE
Option to make --smartcontext drop 25% of context instead of 50%.

### DISCUSSION THREAD

**Discussion by Testertime** [LostRuins/koboldcpp#493] [Ideas]
Hi, I have a small suggestion where I'm very hopeful if you can consider adding it.

Basically, since Llama 2 can use 4096 tokens of context and being able to stretch it by up to 4x (as [explained](https://github.com/LostRuins/koboldcpp/wiki#what-is-rope-config-what-is-ntk-aware-scaling--what-values-to-use-for-rope-config) in your helpful Wiki), the context window is a lot bigger now. When chatting with an AI character, I noticed that the context drop of 50% with smart context can be quite influential on the character's behavior (e.g. when 4096 is cut into 2048).

So I got curious to ask, can you consider adding a setting or parameter to make smart context drop less? (e.g. cutting from 4096 into 3072). I find that smart context is a very valuable feature to prevent constant prompt processing. But I also wished it would keep a bit more context, since in my cases I don't end up filling the emptied context as quickly. It would be awesome if there was an option to customize this.

In case if this is already possible (e.g. by tweaking a file), I would be happy to get a hint where to find it.

Thank you for reading and your amazing project 🙂

**Comment by Z95070** [comment]
Seconding, I swung by to ask if such a thing would be possible and saw this. I often get slightly less than even 50% (3400-3500 tokens) left of a full 8192 history after Smart context is done chopping, and it'd be very useful to be able to crank that up or down at leisure depending on the type of story or chat is going on.
